features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt

```diff
@@ -37,8 +38,11 @@ class DefaultClearCacheUseCase @Inject constructor(
     private val ftueService: FtueService,
     private val pushService: PushService,
     private val seenInvitesStore: SeenInvitesStore,
+    private val activeRoomHolder: ActiveRoomHolder,
 ) : ClearCacheUseCase {
     override suspend fun invoke() = withContext(coroutineDispatchers.io) {
+        // Active rooms should be disposed of before clearing the cache
+        activeRoomHolder.removeRoom(matrixClient.sessionId)?.destroy()
```

Comment:
Now that `activeRoomHolder` stores a list of rooms, I think this code needs to be updated to destroy all the rooms.